code/controllers/subsystem/garbage.dm

```diff
@@ -198,6 +199,8 @@ var/datum/subsystem/garbage_collector/SSgarbage
 					SSgarbage.noqdelhint["[D.type]"] = "[D.type]"
 					testing("WARNING: [D.type] is not returning a qdel hint. It is being placed in the queue. Further instances of this type will also be queued.")
 				SSgarbage.QueueForQueuing(D)
+	else if(D.gc_destroyed == GC_CURRENTLY_BEING_QDELETED)
+		throw EXCEPTION("An items destroy was called multiple times, likely due to a qdel loop in the Destroy logic")
```

Comment:
http://www.byond.com/docs/ref/info.html#/proc/throw It should just crash qdel and continue with Destroy if I'm reading it right